sumo/plotting/dos_plotter.py

```diff
@@ -265,17 +272,30 @@ def get_plot(self, subplot=False, width=None, height=None, xmin=-6.,
         energies = plot_data['energies'][mask]
         fig = plt.gcf()
         lines = plot_data['lines']
-        spins = [Spin.up] if len(lines[0][0]['dens']) == 1 else \
-            [Spin.up, Spin.down]
+        if len(lines[0][0]['dens']) == 1:
+            spins = [Spin.up]
+        elif spin == 'up':
+            spins = [Spin.up]
+        elif spin == 'down':
+            spins = [Spin.down]
+        else:
+            spins = [Spin.up, Spin.down]
 
         for i, line_set in enumerate(plot_data['lines']):
             if subplot:
                 ax = fig.axes[i]
             else:
                 ax = plt.gca()
 
+            ax.set_ylim(plot_data['ymin'], plot_data['ymax'])
+            ax.set_xlim(xmin, xmax)
+
             for line, spin in itertools.product(line_set, spins):
-                if spin == Spin.up:
+                if len(spins) == 1:
+                    label = line['label']
+                    densities = line['dens'][spin][mask]
+                    ax.set_ylim(0, plot_data['ymax'])
```

Comment:
I don't think it makes sense to do this repeatedly inside the loop. We could check after these loops are finished whether there were two spins and set limits accordingly?

Reply:
I've moved the y-axis settings outside the loop now, and checked that it still works the same.